src/notation/view/abstractnotationpaintview.cpp

```diff
@@ -546,7 +546,11 @@ void AbstractNotationPaintView::showElementPopup(const ElementType& elementType,
 
     PopupModelType modelType = AbstractElementPopupModel::modelTypeFromElement(elementType);
 
-    emit showElementPopupRequested(modelType, fromLogical(elementRect).toQRectF());
+    if (elementType == ElementType::DYNAMIC) {
+        emit showElementPopupRequested(modelType, fromLogical(elementRect).toQRect()); // toQRect() is used in AbstractElementPopupModel::updateItemRect()
```

Comment:
I don't quite understand this change. Is it necessary?

Reply:
@cbjeukendrup Initially when we click on dynamic `showElementPopupRequested` is emitted which if you notice uses toQRectF() meaning that the `y` value will be a decimal which will be used to position the popup. But, the `updateItemRect` method (which is called on changing the dynamic) returns a QRect and not QRectF so the value after the decimal will be lost, meaning that it will shift the popup a little higher than initially positioned as seen in below video. (_It's not very noticeable but still..._)

https://github.com/user-attachments/assets/a5ffbb08-a623-4696-b7d4-1763a4a63a9c

I was thinking if we can change the return type of `updateItemRect` to QRectF, so its in sync?

Reply:
Ah, I see! Yes, updating `updateItemRect` and `m_itemRect` to use QRectF seems good to me.